simulator/infra/ports/t/unit_test.cpp

```diff
@@ -132,7 +200,42 @@ int main( int argc, char** argv)
 
 
 namespace ports {
-    namespace testing {
+    namespace testing {  
+
+        bool Logic::check_readiness( uint64 cycle, int check_code, bool is_ready)
+        {
+            assert( check_code >= 0 && check_code <= 3);
+
+            if ( cycle > EXPECTED_MAX_CYCLE)
+                return false;
+
+            if ( check_code == IS_STOP_READY)
+                return (cycle == EXPECTED_MAX_CYCLE) == is_ready;
+            else
```

Comment:
`else` is redundant after `return`

Reply:
Why?

Reply:
If the condition is taken, we leave the function through `return`. Otherwise, we execute everything below. The wide-spread pattern for condition check function:

```c++
if (a)
    return false;

if (b)
    return false;

if (c)
    return false;
// ... a lot of other conditions
return true;
```

It makes the code easy to read (only one level of indentation) and easy to write (you may add or remove conditions just by removing lines)